File: J_se_les12/HW/Core/Logic/Update/UpdateV1.java
package Java.J_se_les12.HW.Core.Logic.Update;

import Java.J_se_les12.HW.Core.Contact.Contact;
import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleInput;
import Java.J_se_les12.HW.Core.Logic.Interfaces.IConsoleOutSuc;
import Java.J_se_les12.HW.Core.Phonebook.Phonebook;

public class UpdateV1 extends UpdateBase {
    IConsoleOutSuc outSuc;

    public UpdateV1(Phonebook phonebook, IConsoleInput input, IConsoleOutSuc outSuc) {
        super(phonebook, input);
        this.outSuc = outSuc;
    }

    @Override
    public void update(int index) {
        if (this.phonebook.sizeOfBook() > 0) {
            Contact contact = phonebook.getContact(index);
            contact.setfName(input.setfName());
            contact.setlName(input.setlName());
            contact.setOrganisation(input.setOrganisation());
            contact.setPhoneNumber(input.setPhoneNumber());
            contact.setNote(input.setNote());
            outSuc.updateSuccess();
        }
    }
}

File: J_se_les13/HW/Core/Logic/Update/UpdateV1.cs
using HW.Core.Phonebook;
using HW.Core.Logic.Interfaces;
using HW.Core.Contact;
namespace HW.Core.Logic.Update
{
    public class UpdateV1 : UpdateBase
    {
        IConsoleOutSuc OutSuc { get; set; }
        public UpdateV1(PhonebookV1 phonebook, IConsoleInput input, IConsoleOutSuc outSuc)
        : base(phonebook, input)
        {
            OutSuc = outSuc;
        }
        public override void Update(int index)
        {
            if (Phonebook.SizeOfBook() > 0)
            {
                ContactV1 tmp = Phonebook.GetContact(index);
                tmp.SetfName(Input.SetfName());
                tmp.SetlName(Input.SetlName());
                tmp.SetOrganisation(Input.SetOrganisation());
                tmp.SetPhoneNumber(Input.SetPhoneNumber());
                tmp.SetNote(Input.SetNote());
                OutSuc.UpdateSuccess();
            }
        }
    }
}